configure.ac

```diff
@@ -114,10 +121,21 @@ AC_ARG_WITH([udevrulesdir],
             [with_udevrulesdir=${libdir}/udev/rules.d])
 AX_NORMALIZE_PATH([with_udevrulesdir])
 AC_SUBST([udevrulesdir], [$with_udevrulesdir])
+
 AC_ARG_WITH([udevrulesprefix],
             [AS_HELP_STRING([--with-udevrulesprefix=XY],[prefix for udev rules file])],
             [AC_SUBST([udevrulesprefix],[$with_udevrulesprefix])])
 AM_CONDITIONAL(WITH_UDEVRULESPREFIX, [test -n "$with_udevrulesprefix"])
+#
+# systemd EnvironmentFile directory
+#
+AC_ARG_WITH([sysdefaultdir],
```

Comment:
Why this is needed?

Reply:
To echo this question and add some detail: does this do anything beyond what the default autotools variables do? I.e. can I just set `--sysconfdir=/etc` at configure time and get the same result?

Reply:
It is needed for tpm2-abrmd.service.in (see discussion above about  @sysdefaultdir@)

Yes you can just set --sysconfdir=/etc, but if you want to do that, why was --with-dbuspolicydir and --with-udevrules-dir added?  Both of those are not really needed if you set --sysconfdir

Also, why do we have --with-systemdsystemunitdir and --with-systempresetdir. Both are not really needed and can be replaced with --libdir

The problem is there are many custom --with-*dir configure aguments added that are not needed.  We should document and use those instead.

But it is weird to have a misture of custom --with-*dir arguments and --libdir/--sysconfdir in order to specify directories.

Reply:
If we choose to not add --with-systemdsystemunitdir and --with-systempresetdir, then we must document that --sysconfdir is needed for these two directories in INSTALL.md.  And we will have inconsistent configure options where some are configured with only --sysconfdir  or --libdir and some are configured with custom configure options.

Reply:
@dantpm not really, for example on Fedora we don't usually install in `/etc/udev/rules.d/` and `/etc/systemd/system/` but instead in `/usr/lib/udev/rules.d` and `/usr/lib/systemd/system`.

But in any case, we will just remove this option right? Since we decided that the field isn't used in the systemd unit file.

Reply:
Yes we will remove the line from the file so the point is moot.

But I am curious.  Are you arguing for more or fewer configure options?
If you want fewer options, the flexibility is not possible for the Fedora example you give.
If you want more options, the flexibility is possible (but requires custom configure options).

Reply:
@dantpm no, I'm arguing for having more standard / consistent / well-known (I don't really know what's the correct term for these) directory variables.

As mentioned by @flihp, these are documented here: https://www.gnu.org/prep/standards/html_node/Directory-Variables.html

The `--with-systemdsystemunitdir` and `--with-udevrulesdir` are not documented there but are de facto standard on systems using udev and systemd. For example in the Fedora package we have macros that set those for us, so we just need to do:

```
%configure --disable-static --disable-silent-rules \
           --with-systemdsystemunitdir=%{_unitdir} \
           --with-udevrulesdir=%{_udevrulesdir}
```

If you add `--with-sysdefaultdir`, then I would do the following in the Fedora package:

```
%configure --disable-static --disable-silent-rules \
           --with-systemdsystemunitdir=%{_unitdir} \
           --with-udevrulesdir=%{_udevrulesdir} \
           --with-sysdefaultdir=%{_sysconfdir}/default
```